### PR TITLE
JanusGraph helm chart - initial commit

### PR DESCRIPTION
**Pull request by samizuh**
**What this PR does / why we need it**:  Helm chart to install JanusGraph. Defaults to standalone experience but can be scaled out and can be configured to use multiple storage and indexing backends.

**Special notes for your reviewer**: @viglesiasce please advise of any changes needed, thx.

### CODE REVIEW

**Comment by samizuh** [comment]
Signed up for CNCF. 

/assign @viglesiasce

File: stable/janusgraph/requirements.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+dependencies:
+  - name: elasticsearch
+    version: 0.4.7
+    repository: http://storage.googleapis.com/kubernetes-charts-incubator
```
Comment (viglesiasce) [inline stable/janusgraph/requirements.yaml]: Use https repo here and use the bucket url scheme as follows:

https://kubernetes-charts-incubator.storage.googleapis.com

**Review by viglesiasce — CHANGES_REQUESTED** [review]

File: stable/janusgraph/requirements.yaml (hunk 1 above)
Comment (samizuh) [reply stable/janusgraph/requirements.yaml]: updated, thx

File: stable/janusgraph/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          resources:
+{{ toYaml .Values.resources | indent 12 }}
+          ports:
+            - containerPort: 8182
+              protocol: TCP
+          volumeMounts:
```
Comment (unguiculus) [inline stable/janusgraph/templates/deployment.yaml:31]: Please make list indentation consistent throughout the chart. To match the prevailing style in other places, indent one more level.

File: stable/janusgraph/templates/pvc.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{{- if and .Values.persistence.enabled (not .Values.persistence.existingClaim) }}
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: {{ template "janusgraph.fullname" . }}
+  labels:
+    app: {{ template "janusgraph.fullname" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
```
Comment (unguiculus) [inline stable/janusgraph/templates/pvc.yaml]: chart: {{ template "janusgraph.chart" . }}

File: stable/janusgraph/templates/pvc.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{{- if and .Values.persistence.enabled (not .Values.persistence.existingClaim) }}
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: {{ template "janusgraph.fullname" . }}
+  labels:
+    app: {{ template "janusgraph.fullname" . }}
```
Comment (unguiculus) [inline stable/janusgraph/templates/pvc.yaml]: app: {{ template "janusgraph.name" . }}

File: stable/janusgraph/values.yaml [hunk 5]
```diff
@@ -0,0 +1,75 @@
+# Default values for JanusGraph chart.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+image:
+  repository: gcr.io/cloud-solutions-images/janusgraph
+  tag: v1
+  pullPolicy: Always
```
Comment (unguiculus) [inline stable/janusgraph/values.yaml]: `IfNotPresent` should be the default.

File: stable/janusgraph/values.yaml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+## set any pod specific resource requests here
+resources: {}
+
+service:
+  type: ClusterIP  # Change to LoadBalancer if you plan to access JanusGraph outside k8s cluster
+  port: 8182
+  serviceAnnotations:
+    # the following line is ignored unless unless using a LoadBalancer with GCP
+    cloud.google.com/load-balancer-type: "Internal"
```
Comment (unguiculus) [inline stable/janusgraph/values.yaml]: Comment this out. It should be optional.

**Review by unguiculus — DISMISSED** [review]

File: stable/janusgraph/templates/pvc.yaml (hunk 3 above)
Comment (samizuh) [reply stable/janusgraph/templates/pvc.yaml]: fixed, thx

File: stable/janusgraph/templates/pvc.yaml (hunk 4 above)
Comment (samizuh) [reply stable/janusgraph/templates/pvc.yaml]: fixed, thx

File: stable/janusgraph/values.yaml (hunk 5 above)
Comment (samizuh) [reply stable/janusgraph/values.yaml]: fixed, thx

File: stable/janusgraph/values.yaml (hunk 6 above)
Comment (samizuh) [reply stable/janusgraph/values.yaml]: fixed, thx

File: stable/janusgraph/templates/deployment.yaml (hunk 2 above)
Comment (samizuh) [reply stable/janusgraph/templates/deployment.yaml:31]: fixed indenting, thx

**Comment by viglesiasce** [comment]
@unguiculus if we have rules on indentation we should add it to the linter. Do you think thats possible?

**Review by viglesiasce — APPROVED** [review]

**Comment by unguiculus** [comment]
@viglesiasce I think we've discussed this. Maybe it was just @mattfarina and me, though. I think we didn't want to add a rule. Some like to indent lists, some don't. We'd cause CI to fail on a lot of charts if we prefer one style over the other. So, I'm just trying to make sure it's consistent within a chart.

**Pull request merged**